### ISSUE TITLE
Filter Spot List by Date

### ISSUE THREAD

**Issue by egbunse** [StraboSpot/strabo-mobile#310]
Add the functionality to see the spots separated out by date edited in the list of all spots.

**Comment by jessicagood** [comment]
Added with commit c38acea. @egbunse Does this give you the functionality you want? It will be in the next release to test, v0.23.0. Feel free to reopen this issue if it's not working right.

There are two date filters. One for the date of the Spot, which initially defaults to the date the Spot is created but the user may edit this field (the field is on the main Spot tab).  The other date field is for the date the Spot was last modified. This is not user-editable.

**Comment by egbunse** [comment]
@jessicagood I have to admit, this wasn't the original way of implementing this we had in mind, but I like this a lot better! We sort of envisioned this becoming date then spots fitting that criteria then the next date in a list if that makes sense which now that I think about it seems clunky compared to filtering by a date interval. So yes, this does accomplish what we wanted! 
However, I am getting an error after updating to 0.23. I already had a dataset pulled up on my Strabo Mobile app so I tested the date filtering with it. But then I logged out to see mess with filtering in another dataset and can log in to another account(s) but can't get the "select an existing project" part to work. I've tried it for two separate accounts with the same results: When I press the "select existing project" button, it spins a long time on "getting projects from server" and then gives me an error message "error communicating with the server; error loading the list of projects from server!" I've tried it on wifi and my 4GLTE connection here in the office, but as I say, I can log in (and see my profile pic), so I think it might be some issue with just the projects part. I'm using a Samsung S7Edge phone and I believe it is up to date on android. Any advice/ideas?
